### PR TITLE
[BPM Abstraction] FWF-1781 dynamic filter implementation

### PR DESCRIPTION
**Pull request by SnehaSuresh-aot**
# Issue Tracking

JIRA: 
Issue Type:FEATURE
https://aottech.atlassian.net/browse/FWF-1781

# Changes

* added dynamic filter implementation in forms-flow-bpm.
* BPM gets filter parameters from UI, and implement the logic to return the tasklist.
* New endpoint created to return the tasklist: `/task-filters`

# How has the change been tested?

# Screenshots (if applicable)
image

# Build Success screenshot (Till a CICD pipeline is set up)

# Notes

### CODE REVIEW

File: forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/FilterRestResource.java [hunk 1]
```diff
@@ -1,48 +1,29 @@
...
-    List<FilterDto> getFilters(@Context UriInfo uriInfo, @QueryParam("itemCount") Boolean itemCount,
-                                          @QueryParam("firstResult") Integer firstResult, @QueryParam("maxResults") Integer maxResults);
-
-    @GET
-    @Path("/{id}/list")
-    @Produces({MediaType.APPLICATION_JSON, Hal.APPLICATION_HAL_JSON})
-    Object executeList(@Context Request request, @QueryParam("firstResult") Integer firstResult, @QueryParam("maxResults") Integer maxResults, @PathParam("id") String id);
-
     @POST
-    @Path("/{id}/list")
```
Comment (sumesh-aot) [inline forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/FilterRestResource.java:37]: Can we do this is in a way to not break existing implementations if we merge the code. I mean keep existing endpoints for filters there and mark as deprecated. Then create a new resource for new implementation? If the name `/filters/tasks` is causing issue, let's create another name like `task-filters`. What do you think ?

**Review by sumesh-aot — COMMENTED** [review]
Please add some unit test cases

File: forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/FilterRestResource.java (hunk 1 above)
Comment (sreehari-aot) [reply forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/FilterRestResource.java:37]: I agree, if it's the new implementation for existing endpoints a new versioned endpoint would be ideal, `/v2/filters/tasks`. I am not sure if it's the case here.

Comment (shibin-aot) [reply forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/FilterRestResource.java:37]: Better to create a v2 resource and add the new endpoints in that. That was the idea when we designed this pattern. Also, mark the v1 resource as deprecated would be a good idea.
NOTE: this need to be done only if there is a drastic change in the existing resource / implementation. If the purpose is just to create a new api which does something else then go with v1 itself.

Comment (sumesh-aot) [reply forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/FilterRestResource.java:37]: It's not really v2 of existing endpoints, rather a new implementation by not using any of the underlying filter implementation. We could go with v2 though, but doesn't really align with typical versioning.

Comment (SnehaSuresh-aot) [reply forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/FilterRestResource.java:37]: Yeah I agree, I think we can go with v1 and add a new resource for new implementation. Also existing implementation can be deprecated.

**Review by sreehari-aot — APPROVED** [review]
LGTM!

**Review by abilpraju-aot — APPROVED** [review]

**Review by abhilash-aot — APPROVED** [review]

**Pull request merged**